simulariumio/writers/binary_writer.py

```diff
@@ -0,0 +1,441 @@
+            )
+        return result
+
+    @staticmethod
+    def _header_n_bytes() -> int:
+        """
+        Get length of binary header in bytes
+        """
+        return (
+            len(BINARY_SETTINGS.FILE_IDENTIFIER)
+            + BINARY_SETTINGS.BYTES_PER_VALUE * BINARY_SETTINGS.HEADER_N_INT_VALUES
+        )
+
+    @staticmethod
+    def _trajectory_info_length(
+        trajectory_data: TrajectoryData,
+        type_mapping: Dict[str, Any],
+    ) -> int:
+        """
+        Get length of trajectory info in JSON
+        (n_bytes = n_values)
+        """
+        total_steps = trajectory_data.agent_data.total_timesteps()
+        traj_info = Writer._get_trajectory_info(
+            trajectory_data, total_steps, type_mapping
+        )
+        traj_info_n_bytes = (
+            BINARY_SETTINGS.BLOCK_HEADER_N_VALUES * BINARY_SETTINGS.BYTES_PER_VALUE
+            + len(json.dumps(traj_info))
+        )
+        return traj_info_n_bytes + BinaryWriter._padding(traj_info_n_bytes)
+
+    @staticmethod
+    def _plot_data_length(plots: List[Dict[str, Any]]) -> int:
+        """
+        Get length of plot data in JSON
+        (n_bytes = n_values)
+        """
+        plot_data_n_bytes = (
+            BINARY_SETTINGS.BLOCK_HEADER_N_VALUES * BINARY_SETTINGS.BYTES_PER_VALUE
+            + len(
+                json.dumps(
+                    {
+                        "version": CURRENT_VERSION.PLOT_DATA,
+                        "data": plots,
+                    },
+                )
+            )
+        )
+        return plot_data_n_bytes + BinaryWriter._padding(plot_data_n_bytes)
+
+    @staticmethod
+    def _chunk_files(
+        trajectory_data: TrajectoryData,
+        type_mapping: Dict[str, Any],
+        frame_buffers_n_values: List[int],
+        max_bytes: int,
+    ) -> Tuple[List[BinaryChunk], int, int]:
+        """
+        Get number of frames, number of bytes, and number of values
+        for each file that will be written,
+        multiple files if needed to satisfy file size limits
+        also return size of trajectory info and plot data
+        """
+        header_n_bytes = BinaryWriter._header_n_bytes()
+        traj_info_n_bytes = BinaryWriter._trajectory_info_length(
+            trajectory_data, type_mapping
+        )
+        plot_data_n_bytes = BinaryWriter._plot_data_length(trajectory_data.plots)
+        max_spatial_bytes = (
+            max_bytes - header_n_bytes - traj_info_n_bytes - plot_data_n_bytes
+        )
+        file_chunks = [BinaryChunk()]
+        current_chunk = 0
+        current_frames_bytes = 0
+        for frame_index, buffer_size in enumerate(frame_buffers_n_values):
+            spatial_header_n_values = (
+                BINARY_SETTINGS.BLOCK_HEADER_N_VALUES
+                + BINARY_SETTINGS.SPATIAL_BLOCK_HEADER_CONSTANT_N_VALUES
+                + 2 * (file_chunks[current_chunk].n_frames + 1)
```

Comment:
is there already a symbolic constant for this `2`?

Reply:
no but I'll add one